upsolver-ui/src/components/LoginForm/LoginForm.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useContext } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { UserContext } from "../../UserContext.js";
+import "./LoginForm.css";
+import Logo from "../Logo/Logo";
+import { Paper, Box, Typography, TextField, Button } from "@mui/material";
+
+export default function LoginForm() {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const { updateUser } = useContext(UserContext);
+
+  const handleLogin = async (e) => {
+    try {
+      e.preventDefault();
+      const response = await fetch("http://localhost:3001/users/login", {
```

Comment:
For testing this is ok, but for the long-term I'd recommend storing the base URL somewhere as a constant:
```
const BASE_URL = 'http://localhost:3001;
```

You could even look into detecting if you are in development or production mode as a follow-up, so if you have a live URL (e.g. hosted through your github), you can reference that when running production. You could look into the `__DEV__` variable or some similar mechanism to detect the mode that the app is running in.

Reply:
Solved in: [https://github.com/cdamezcua/Upsolver/pull/19](url)